lib/decoder.ex

```diff
@@ -86,15 +86,18 @@ defmodule Jason.Decoder do
     end
   end
 
-  defp float_decode_function(%{floats: :decimals}) do
-    fn string, token, skip ->
-      # silence xref warning
-      decimal = Decimal
-      try do
-        decimal.new(string)
-      rescue
-        Decimal.Error ->
-          token_error(token, skip)
+  if Code.ensure_loaded?(Decimal) do
+    defp float_decode_function(%{floats: :decimals}) do
+      fn string, token, skip ->
+        # silence xref warning
+        decimal = Decimal
+
+        try do
+          decimal.new(string)
+        rescue
+          Decimal.Error ->
+            token_error(token, skip)
+        end
```

Comment:
I think at this point, we should just remove the old work-around for silencing the warnings
```suggestion
  if Code.ensure_loaded?(Decimal) do
    defp float_decode_function(%{floats: :decimals}) do
      fn string, token, skip ->
        try do
          Decimal.new(string)
        rescue
          Decimal.Error ->
            token_error(token, skip)
        end
```

Reply:
Additionally, do you think it would make sense to have a function that would emit a nice error if this option is passed but decimal was not included? I think this would be a good usability upgrade.

Reply:
> I think at this point, we should just remove the old work-around for silencing the warnings

Oh good. I *almost* did that and then wasn't 100% sure there wasn't some weird edge case. I added a commit to the PR that removes them.

Reply:
> Additionally, do you think it would make sense to have a function that would emit a nice error if this option is passed but decimal was not included? I think this would be a good usability upgrade.

Yes, that seems nice. I raised in the decoder with a message I think is consistent with the library. The encoder calls seemed fine since they required `Decimal` structs to be passed.